snakemake_executor_plugin_slurm/__init__.py

```
@@ -107,6 +108,20 @@
     def additional_general_args(self):
         return "--executor slurm-jobstep --jobs 1"
 
+    def run_jobs(self, jobs: List[JobExecutorInterface]):
+        for _, same_rule_jobs in groupby(jobs, key=lambda job: job.rule.name):
+            if len(same_rule_jobs) == 1:
+                self.run_job(same_rule_jobs[0])
```

Ruff (0.8.0)

Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage

(B031)